### PR TITLE
Big oops updates

### PR DESCRIPTION
**Pull request by markshowalter**
Here are the big updates we've discussed. Most are described in the slides I shared a week ago. A few other notes:
1-Floating point array compression has been sitting around in one of my local polymath branches for a while, as part of our support for `__getstate__/__setstate__`, aka serialization, aka "pickle". I cleaned that up, added a bunch of unit tests, and it is now the one big change to polymath included here. It uses fpzip (a very impressive module, now added to requirements.txt) in addition to some other techniques. This is important because our still-mythical online backplane service will need to be able to store big data arrays efficiently, and this appears to make that possible. Basically, you can specify how many digits of precision any Qube object needs to preserve, and it will find an efficient way to achieve that when it is pickled. fpzip takes advantage of the fact that our backplane arrays vary smoothly to achieve very impressive reductions in their storage requirements.
2-There are a few new methods here and there with names that end with "_experimental". They are not currently used, but I have been testing them. I plan to create a GitHub issue for each of them. For now, you can ignore.
3-I had to tweak the JunoCam module because its method of defining a cadence using a tuple of input parameters is no longer allowed--part of the transition to the new TimedImage subclass of Observation.
4-One (at least) of the backplanes now fails for both JunoCam and JIRAM. Joe and I might need to work together to track that down. All other unit tests pass.
5-FYI, I am not sure I spoke correctly recently when I said that backplanes always come with shapes that are broadcastable to their data arrays. That was not originally the case when I started writing the backplane module long ago, and it does not appear that it was ever fully implemented. I _do_ think it is how backplanes should work going forward, but for now, caveat emptor.
6-Please ignore hosts/jwst. It's a work in progress.

### CODE REVIEW

**Review by rfrenchseti — CHANGES_REQUESTED** [review]
- Runs my navigation code with no errors and no change in output (using the old classes).
- fpzip doesn't work with Python 3.11; all tests done with 3.10
- Rather than commenting out pushframe, pushbroom, rasterscan, rasterslit, slit, move that code to the archive directory
- Should the .txt and .fits files in hosts.solar have more recognizable names associated with the particular .py file that reads them?
- hosts.solar needs unit tests; perhaps create an issue for this

**Review by jnspitale — APPROVED** [review]
Many JIRAM IMG backplane unit tests fail in this pull request because their results do not match the reference backplanes.  However, all indications are that a previously unknown problem was fixed here.  

First, these tests can be made to fail in the master branch by removing the preceding backplane exercise, suggesting that something was left in an invalid state by the preceding test, resulting in a bad reference backplane for the subsequent test.  Second, in the master branch, the reference backplane generated (prior to bytescaling) for the first failed test (Right ascension (deg, actual)) clearly does not correspond to valid angles in radians.  Finally, the backplane generated in the pull request matches OMINAS' results to pretty good precision.  Therefore the failures are due to bad reference backplanes.

OMINAS does not currently contain all of the backplane calculations that would be needed to validate every one of these failed backplanes, but based on the above observations, I'm approving the pull request.

File: oops/cadence/sequence.py [hunk 1]
```diff
@@ -308,346 +382,48 @@ def runTest(self):
...
-        self.assertTrue((abs(time_shifted-time-1.) < 1.e-13).all())
-
-        ####################################
-        # Converted-to-continuous case
-        # We just do spot-checking here
-        ####################################
-
-        cadence = cadence.as_continuous()
-        self.assertTrue(cadence.is_continuous)
+        # cadence = Metronome(100., 10., 7.5, 4)
```
[the remark this answers is not in the document]
Comment (markshowalter) [reply oops/cadence/sequence.py:402]: I added a bit to the comment, but yes. What I'm doing here is importing a unit test from Metronome, and then showing that it yields the exact same results (but less efficiently) if we define a Sequence with identical properties. A lot of the cadence unit tests employ this approach.

File: requirements.txt [hunk 2]
```diff
@@ -1,4 +1,5 @@
 cspyce>=2.0.7
+fpzip>=1.2
```
[the remark this answers is not in the document]
Comment (markshowalter) [reply requirements.txt:2]: astropy added.

**Pull request merged**